spec/Memory_Consistency.tex

```diff
@@ -268,72 +278,79 @@ \section{Sequential Consistency for Data Race Free Programs}
 % \end{tabular}
 % \end{center}
 
-\section{Data Structure Consistency}
-
-Generally speaking, data structures provided by the Chapel language and
-libraries are not safe for concurrent use when one of those uses is
-modification. In particular, 
-
-\begin{itemize}
-
-\item Elements of a Chapel array are not safe against \textit{data races}
-unless they are of an \textit{atomic} or \textit{sync} type.
-
-\item Chapel arrays and domains are not safe for concurrent changes to
-their shape.
-
-\item Chapel arrays are safe for concurrent updates to different elements
-that already exist or for read-only access from multiple tasks.
-
-\end{itemize}
-
-Data structure authors should make an effort to document how their data
-structure is safe or is not safe for concurrent updates. Generally speaking,
-data structures might fall into these categories:
-
-\begin{itemize}
-
-\item safe for any concurrent updates (e.g. a work queue)
-
-\item safe for concurrent updates to disjoint regions but not for shape
-changes (e.g. an array)
-
-\item not safe for any concurrent updates, even to disjoint regions (e.g.
-the a string that reallocates itself when a character is changed)
-
-\end{itemize}
-
-The data structure consistency could operate with either of these
-strategies:
 
-\begin{itemize}
-
-\item data structure updates will be visible once the usual task
-synchronization takes place, or
-
-\item the data structure requires consistency to be managed explicitly
-(e.g. with a 'flush' call).
-
-\end{itemize}
+% I don't think this belongs in the MCM section. I think it's valuable, but
+% inappropriate for it to be here in it's current form. I'm guessing this is
+% meant to lead into some of the discussions we had around optimizing data
+% structures and allowing them to be invalid until a "flush", but it's current
+% form talks about implementation specific aspects of our data structures.
+%\section{Data Structure Consistency}
```

Comment:
That's fine, but we have to document this somewhere. Where else can we write this information? In the section on Arrays, for example? Arrays and Domains are Chapel-level enough that I think they need to be covered in the spec. If they're covered in the spec, I'd expect the Memory Consistency section to at least point to the answer of how they can be used in a parallel way...